### ISSUE TITLE
Crash when user formula z^3 + c^2*z^2 + c

### ISSUE THREAD

**Issue by jblang** [xaos-project/XaoS#56]
_From [matthieu...@gmail.com](https://code.google.com/u/107870813173416775174/) on December 14, 2011 07:06:21_

Xaos 3.5, running on Windows 7, crashes consistently with user formula z^3 + c^2*z^2 + c on Mandelbrot mode.

I tried changing slightly the formula, it appears to crash for any formula of the form z^3 + (function of c) z^2 + c, which is problematic.

Also, there is no problem without the z^2 term.

_Original issue: http://code.google.com/p/gnuxaos/issues/detail?id=57_

**Comment by jblang** [comment]
Reproduced on master. Valgrind stack trace:

```
==12216== Invalid read of size 1
==12216==    at 0x1808F3: sffe_parse (sffe.c:687)
==12216==    by 0x126A9F: uih_sffein (menu.c:1407)
==12216==    by 0x112CD6: ui_menuactivate (ui.c:415)
==12216==    by 0x11ECE1: ui_dialogmouse (dialog.c:1421)
==12216==    by 0x111C0C: ui_mouse (ui.c:643)
==12216==    by 0x11058C: ui_mainloop (ui.c:1671)
==12216==    by 0x11058C: main (ui.c:1482)
==12216==  Address 0x72b7320 is 0 bytes after a block of size 16 alloc'd
==12216==    at 0x4C31D2F: realloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==12216==    by 0x1809B0: sffe_parse (sffe.c:689)
==12216==    by 0x126A9F: uih_sffein (menu.c:1407)
==12216==    by 0x112CD6: ui_menuactivate (ui.c:415)
==12216==    by 0x11ECE1: ui_dialogmouse (dialog.c:1421)
==12216==    by 0x111C0C: ui_mouse (ui.c:643)
==12216==    by 0x11058C: ui_mainloop (ui.c:1671)
==12216==    by 0x11058C: main (ui.c:1482)
==12216== 
==12216== Invalid read of size 8
==12216==    at 0x180984: sffe_parse (sffe.c:688)
==12216==    by 0x126A9F: uih_sffein (menu.c:1407)
==12216==    by 0x112CD6: ui_menuactivate (ui.c:415)
==12216==    by 0x11ECE1: ui_dialogmouse (dialog.c:1421)
==12216==    by 0x111C0C: ui_mouse (ui.c:643)
==12216==    by 0x11058C: ui_mainloop (ui.c:1671)
==12216==    by 0x11058C: main (ui.c:1482)
==12216==  Address 0x72b7328 is 8 bytes after a block of size 16 alloc'd
==12216==    at 0x4C31D2F: realloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==12216==    by 0x1809B0: sffe_parse (sffe.c:689)
==12216==    by 0x126A9F: uih_sffein (menu.c:1407)
==12216==    by 0x112CD6: ui_menuactivate (ui.c:415)
==12216==    by 0x11ECE1: ui_dialogmouse (dialog.c:1421)
==12216==    by 0x111C0C: ui_mouse (ui.c:643)
==12216==    by 0x11058C: ui_mainloop (ui.c:1671)
==12216==    by 0x11058C: main (ui.c:1482)
==12216== 
==12216== Jump to the invalid address stated on the next line
==12216==    at 0x0: ???
==12216==    by 0x17F11C: sffe_eval (sffe.c:152)
==12216==    by 0x14C875: sffe_calc (docalc.c:271)
==12216==    by 0x16B252: calculate (calculate.h:19)
==12216==    by 0x16B252: calculatepixel (btrace.c:240)
==12216==    by 0x16BAF3: tracecolor32 (btraced.c:43)
==12216==    by 0x16BAF3: tracerectangle (btrace.c:430)
==12216==    by 0x16E571: boundarytrace (btrace.c:503)
==12216==    by 0x165CA7: do_fractal (zoom.c:1496)
==12216==    by 0x121C53: uih_do_fractal (ui_helper.c:987)
==12216==    by 0x110794: ui_mainloop (ui.c:1641)
==12216==    by 0x110794: main (ui.c:1482)
==12216==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==12216== 
==12216== 
==12216== Process terminating with default action of signal 11 (SIGSEGV)
==12216==  Bad permissions for mapped region at address 0x0
==12216==    at 0x0: ???
==12216==    by 0x17F11C: sffe_eval (sffe.c:152)
==12216==    by 0x14C875: sffe_calc (docalc.c:271)
==12216==    by 0x16B252: calculate (calculate.h:19)
==12216==    by 0x16B252: calculatepixel (btrace.c:240)
==12216==    by 0x16BAF3: tracecolor32 (btraced.c:43)
==12216==    by 0x16BAF3: tracerectangle (btrace.c:430)
==12216==    by 0x16E571: boundarytrace (btrace.c:503)
==12216==    by 0x165CA7: do_fractal (zoom.c:1496)
==12216==    by 0x121C53: uih_do_fractal (ui_helper.c:987)
==12216==    by 0x110794: ui_mainloop (ui.c:1641)
==12216==    by 0x110794: main (ui.c:1482)
==12216== 
==12216== HEAP SUMMARY:
==12216==     in use at exit: 3,334,449 bytes in 2,477 blocks
==12216==   total heap usage: 6,799 allocs, 4,322 frees, 37,350,880 bytes allocated
==12216== 
==12216== LEAK SUMMARY:
==12216==    definitely lost: 329 bytes in 8 blocks
==12216==    indirectly lost: 0 bytes in 0 blocks
==12216==      possibly lost: 0 bytes in 0 blocks
==12216==    still reachable: 3,334,120 bytes in 2,469 blocks
==12216==         suppressed: 0 bytes in 0 blocks
==12216== Rerun with --leak-check=full to see details of leaked memory
==12216== 
==12216== For counts of detected and suppressed errors, rerun with: -v
```

**Comment by jblang** [comment]
Fixed by updating to latest sffe code in cf5befe412cd78fa0c2f1ae058084c79c081be8e
